@here/olp-sdk-dataservice-read/lib/client/SubscribeRequest.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { StreamApi } from "@here/olp-sdk-dataservice-api";
+
+/**
+ * Prepares information for calls to the OLP Stream Service.
+ */
+export class SubscribeRequest {
+    private mode?: "serial" | "parallel";
+    private subscriptionId?: string;
+    private consumerId?: string;
+    private subscriptionProperties?: StreamApi.SubscriptionProperties;
+
+    /**
+     * Gets the subscription mode for the request.
+     *
+     * @return The subscription mode.
+     */
+    public getMode(): "serial" | "parallel" | undefined {
+        return this.mode;
+    }
+
+    /**
+     * A setter for the provided subscription mode.
+     *
+     * @param mode The subscription mode.
+     * @returns The [[SubscribeRequest]] instance that you can use to chain methods.
+     */
+    public withMode(mode: "serial" | "parallel"): SubscribeRequest {
+        this.mode = mode;
+        return this;
+    }
+
+    /**
+     * Gets the subscription id for the request.
+     *
+     * @return The subscription id.
+     */
+    public getSubscriptionId(): string | undefined {
+        return this.subscriptionId;
+    }
+
+    /**
+     * A setter for the provided subscription id.
+     *
+     * @param id The subscription id.
+     * @returns The [[SubscribeRequest]] instance that you can use to chain methods.
+     */
+    public withSubscriptionId(id: string): SubscribeRequest {
+        this.subscriptionId = id;
+        return this;
+    }
+
+    /**
+     * Gets the consumer id for the request.
+     *
+     * @return The consumer id.
+     */
+    public getConsumerId(): string | undefined {
+        return this.consumerId;
+    }
+
+    /**
+     * A setter for the provided consumer id.
+     *
+     * @param id The consumer id.
+     * @returns The [[SubscribeRequest]] instance that you can use to chain methods.
+     */
+    public withConsumerId(id: string): SubscribeRequest {
+        this.consumerId = id;
+        return this;
+    }
+
+    /**
+     * Gets the subscriptionProperties for the request - one or more Kafka Consumer properties to use for this subscription.
+     *
+     * @return The subscriptionProperties.
+     */
+    public getSubscriptionProperties():
+        | StreamApi.SubscriptionProperties
```

Comment:
I would not do it like this.
I would allow users to specify key/value as string.
Same as we did for the C++ SDK would be my preferred way. 
Else you risk breaking API every time Kafka adds/changes a new parameter.

Reply:
Replaced